src/jobs/optimize-image.ts

```diff
@@ -40,42 +42,53 @@ const optimizeImage = async (job: Job) => {
 
   try {
     const profiler = logger.startTimer();
-    logger.info(
-      `MinIO is at ${MINIO_HOST}:${MINIO_API_PORT} ${MINIO_ROOT_USER}`
-    );
 
     logger.info(`Starting to optimize images ${model}/${destinationId}`);
-    const { buffer, size } = await getBufferFromMinio(
+    const { buffer } = await getBufferFromStorage(
       minioClient,
       incomingMinioBucket,
-      destinationId,
-      logger
+      destinationId
     );
+
     await createBucketIfNotExists(minioClient, finalMinioBucket, logger);
 
-    logger.info(`Got object of size ${size}`);
+    // logger.info(`Got object of size ${size}`);
     const promises = Object.entries(config)
-      .map(([key, value]) => {
-        const outputType = key as "webp" | "jpeg" | "png"; // output type (jpeg, webp)
+      .map(([key, value], i) => {
+        const outputType = key as "webp"; // output type (jpeg, webp)
         const {
           // @ts-ignore
           options = {},
-          // @ts-ignore
           variants = [],
           // @ts-ignore
           ext = defaultOptions[outputType].ext,
-        } = value;
+        } = value as {
+          options: {
+            [key: string]: unknown;
+          };
+          variants: { width: number; height: number }[];
+        };
 
         return variants.map(
-          async (variant: {
-            extract?: any;
-            resize?: any;
-            outputOptions?: any;
-            blur?: any;
-            width?: number;
-            height?: number;
-            suffix?: any;
-          }) => {
+          async (
+            variant: {
+              extract?: any;
+              resize?: any;
+              outputOptions?: any;
+              blur?: any;
+              width?: number;
+              height?: number;
+              suffix?: any;
+            },
+            j
+          ) => {
+            // This is a pretty hacky way of sleeping and probably something we shoudl
+            // not do?
+            // But basically the reason we have to do this is because backblaze
+            // seems to not like it when we bombard it with files all in one go,
+            // so when we do Promise.all at the end of this we're sending all the
+            // files and it just doesn't process anything but the first one.
+            await sleep((i + 1) * (j + 1) * 1500);
```

Comment:
Having to sleep in these promises because if I don't then backblaze seems to get overwhelmed and returns an error 408